### DISCUSSION TITLE
Problem updating a problem from old reaktoro version to the newest version

### DISCUSSION THREAD

**Discussion by JMCarmona10** [reaktoro/reaktoro#331] [General]
Hello Lean,

Lately, I've been facing a challenge concerning the transport-reaktoro functionality from the older Reaktoro version, as outlined in this link (https://reaktoro.org/v1/tutorials/transport/reactive-transport-calcite-dolomite-custom.html). I've been working on updating this program with the latest function calls, but I've encountered difficulties when defining temperature and pressure parameters.

It seems that in the new Reaktoro version, the 'simulate()' function no longer recognizes the 'set.temperature' parameter. Do you have any suggestions on how to resolve this issue?

![reaktoro_problem](https://github.com/reaktoro/reaktoro/assets/116467897/309790d0-8b1b-48e9-b7dd-550b0dea63cb)

This is the section where I'm encountering difficulties. I've attempted to upgrade this program using the latest Reaktoro functions. However, when I run the program, I encounter an error when defining the temperature (T) and pressure (P) for that composition as shown in the figure above. I could show the entire program but it too long. 

`def simulate():

    # Step 7.1: Construct the chemical system with its phases and species
    db = SupcrtDatabase("supcrtbl")

    brine = ([
        "H2O(aq)",
        "H+",
        "OH-",
        "Na+",
        "Cl-",
        "Ca+2",
        "Mg+2",
        "HCO3-",
        "CO2(aq)",
        "CO3-2"])

    solution = AqueousPhase(brine)
    solution.set(ActivityModelPitzer())

    # aqueous species are individually selected for performance reasons
    mineral = MineralPhase("Calcite Dolomite Quartz")

    # Step 7.2: Create the ChemicalSystem object using the configured editor
    system = ChemicalSystem(db, solution, mineral)

    # Step 7.3: Define the initial condition of the reactive transport modeling problem
    problem_ic = EquilibriumProblem(system)
    problem_ic.setTemperature(T)
    problem_ic.setTemperature(P)
    #problem_ic.setTemperature(333.15, "celsius")
    #problem_ic.setPressure(10, "atm")
    problem_ic.add("H2O(aq)", 1.0, "kg")
    problem_ic.add("NaCl", 0.7, "mol")
    problem_ic.add("CaCO3", 10, "mol")
    problem_ic.add("SiO2", 10, "mol")`
`
`

**Comment by allanleal** [comment]
Hi @JMCarmona10 , please do not use `EquilibriumProblem`. It's meant for another purpose in Reaktoro v2. You need to use `Material` for allowing you to use chemical formulas instead of exact species names that exist in the system.

**Reply by JMCarmona10** [reply]
Hello, @allanleal  
I appreciate your guidance. I have encountered another issue. Within a particular segment of the program, I find myself needing to define the  species in the fluid phase. I attempted to utilize the "speciesWithAggregateState()" function, but I have discovered that this function is operational solely within the database, and not within the system. If my objective is to ascertain the quantity of species within the liquid phase, I would greatly value your recommendation on an alternative approach.


![second error](https://github.com/reaktoro/reaktoro/assets/116467897/ff540ebb-4a37-4b67-9623-359921e301b1)

**Comment by allanleal** [comment]
Better to initiate a new question with a new title next time. 

`system.species()` return a `SpeciesList` object. This class has a method `withAggregateState` which return a new `SpeciesList` object with species with the aggregate state of interest:

~~~py
aqueous_species = system.species().withAggregateState(AggregateState.Aqueous)
gaseous_species = system.species().withAggregateState(AggregateState.Gas)
~~~

See API here: https://reaktoro.org/api/classReaktoro_1_1SpeciesList.html

**Comment by JMCarmona10** [comment]
Hello Allan, I am once again trying to update this problem in Reaktoro v1. In the original code, there is a section "Define an auxiliary function for creating the output file" that uses the ChemicalOutput class:

output = ChemicalOutput(system)
output.filename('results/{}.txt'.format(str(step).zfill(ndigits)))

However, I dont know which class to use i think its ChemicalState class instead but im unsure. 

state = ChemicalState(system)

Also i dont know how to use Reaktoro API to re-write this particular line. 

"output.filename('results/{}.txt'.format(str(step).zfill(ndigits)))"

I would really appreciate if you could provide an example of the Reaktoro API syntax needed to replace that output.filename() line and ChemicalOutput.   Here is the section that im trying to figure `   
   
  # Create the instance of ChemicalOutput class
        output = ChemicalOutput(system)

        # The number of digits in number of steps (e.g., 100 has 3 digits)
        ndigits = len(str(nsteps))

        # Provide the output file name, which will correspond
        output.filename('results/{}.txt'.format(str(step).zfill(ndigits)))

        # We define the columns' tags filled with the name of the quantities
        # The first column has a tag 'x' (which corresponds to the center coordinates of the cells )
        output.add('tag', 'x') # The value of the center coordinates of the cells

        # The rest of the columns correspond to the requested properties
        for quantity in output_quantities:
            output.add(quantity)

        # We update the file with states that correspond to the cells' coordinates stored in x
        output.open()
        for state, tag in zip(states, x):
            output.update(state, tag)
        output.close()`

**Reply by allanleal** [reply]
Reaktoro v2 does not have a `ChemicalOutput` class. You'll need to get the properties of interest from a `ChemicalProps` object.
